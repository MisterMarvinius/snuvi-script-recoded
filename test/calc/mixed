print(2 + 3 * 4);
print((2 + 3) * 4);
print(2 + 3 * 4 + 2);
print(2 + 3 * (4 + 2));
print(2 + 3 * (4 + 2));
print(2 + -3 * (4 + 2));
print(2 + 3 * (-4 + 2));
print((2 - 3) * 4);
