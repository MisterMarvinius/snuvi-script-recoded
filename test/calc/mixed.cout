push 2
push 3
push 4
use *(2)
use +(2)
use print(1)
push 2
push 3
use +(2)
push 4
use *(2)
use print(1)
push 2
push 3
push 4
use *(2)
use +(2)
push 2
use +(2)
use print(1)
push 2
push 3
push 4
push 2
use +(2)
use *(2)
use +(2)
use print(1)
push 2
push 3
push 4
push 2
use +(2)
use *(2)
use +(2)
use print(1)
push 2
push 3
use -(1)
push 4
push 2
use +(2)
use *(2)
use +(2)
use print(1)
push 2
push 3
push 4
use -(1)
push 2
use +(2)
use *(2)
use +(2)
use print(1)
push 2
push 3
use -(2)
push 4
use *(2)
use print(1)
