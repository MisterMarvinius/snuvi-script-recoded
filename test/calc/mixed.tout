(1, LITERAL, "print")
(1, OPEN_BRACKET)
(1, NUMBER, 2.0)
(1, ADD)
(1, NUMBER, 3.0)
(1, MUL)
(1, NUMBER, 4.0)
(1, CLOSE_BRACKET)
(1, SEMICOLON)
(2, LITERAL, "print")
(2, OPEN_BRACKET)
(2, OPEN_BRACKET)
(2, NUMBER, 2.0)
(2, ADD)
(2, NUMBER, 3.0)
(2, CLOSE_BRACKET)
(2, MUL)
(2, NUMBER, 4.0)
(2, CLOSE_BRACKET)
(2, SEMICOLON)
(3, LITERAL, "print")
(3, OPEN_BRACKET)
(3, NUMBER, 2.0)
(3, ADD)
(3, NUMBER, 3.0)
(3, MUL)
(3, NUMBER, 4.0)
(3, ADD)
(3, NUMBER, 2.0)
(3, CLOSE_BRACKET)
(3, SEMICOLON)
(4, LITERAL, "print")
(4, OPEN_BRACKET)
(4, NUMBER, 2.0)
(4, ADD)
(4, NUMBER, 3.0)
(4, MUL)
(4, OPEN_BRACKET)
(4, NUMBER, 4.0)
(4, ADD)
(4, NUMBER, 2.0)
(4, CLOSE_BRACKET)
(4, CLOSE_BRACKET)
(4, SEMICOLON)
(5, LITERAL, "print")
(5, OPEN_BRACKET)
(5, NUMBER, 2.0)
(5, ADD)
(5, NUMBER, 3.0)
(5, MUL)
(5, OPEN_BRACKET)
(5, NUMBER, 4.0)
(5, ADD)
(5, NUMBER, 2.0)
(5, CLOSE_BRACKET)
(5, CLOSE_BRACKET)
(5, SEMICOLON)
(6, LITERAL, "print")
(6, OPEN_BRACKET)
(6, NUMBER, 2.0)
(6, ADD)
(6, SUB)
(6, NUMBER, 3.0)
(6, MUL)
(6, OPEN_BRACKET)
(6, NUMBER, 4.0)
(6, ADD)
(6, NUMBER, 2.0)
(6, CLOSE_BRACKET)
(6, CLOSE_BRACKET)
(6, SEMICOLON)
(7, LITERAL, "print")
(7, OPEN_BRACKET)
(7, NUMBER, 2.0)
(7, ADD)
(7, NUMBER, 3.0)
(7, MUL)
(7, OPEN_BRACKET)
(7, SUB)
(7, NUMBER, 4.0)
(7, ADD)
(7, NUMBER, 2.0)
(7, CLOSE_BRACKET)
(7, CLOSE_BRACKET)
(7, SEMICOLON)
(8, LITERAL, "print")
(8, OPEN_BRACKET)
(8, OPEN_BRACKET)
(8, NUMBER, 2.0)
(8, SUB)
(8, NUMBER, 3.0)
(8, CLOSE_BRACKET)
(8, MUL)
(8, NUMBER, 4.0)
(8, CLOSE_BRACKET)
(8, SEMICOLON)
