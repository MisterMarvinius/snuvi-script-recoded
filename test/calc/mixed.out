14.0
20.0
16.0
20.0
20.0
-16.0
-4.0
-4.0
